app/controllers/application_controller.rb

```diff
@@ -88,7 +88,14 @@ def set_locale
 
   def redirect_unless_admin
     unless current_user.admin?
-      redirect_to stream_url, :notice => 'you need to be an admin to do that'
+      redirect_to stream_url, notice: "you need to be an admin to do that"
+      return
+    end
+  end
+
+  def redirect_unless_admin_or_moderator
+    unless current_user.moderator? || current_user.admin?
```

Comment:
@troessner or @jhass could someone of you explain how that guard clause should look like? We didn't manage to change the code accordingly.